src/platform/packages/shared/kbn-scout/src/cli/test_filtering.ts

```diff
@@ -0,0 +1,402 @@
+        } else if (execError.output && execError.output[1]) {
+          output =
+            typeof execError.output[1] === 'string'
+              ? execError.output[1]
+              : execError.output[1].toString();
+        } else {
+          // If no stdout, log the error and continue to next config
+          log.warning(
+            `Playwright command failed for ${pwConfigPath}: ${execError.message}. Skipping...`
+          );
+          continue;
+        }
+      }
+
+      // Parse the JSON output once in memory
+      const specs = parsePlaywrightOutput(output, pwConfigPath, log);
+      if (!specs) {
+        continue;
+      }
+
+      // Check all tags
+      const foundTags = new Set<string>();
+      for (const tag of tags) {
+        const hasTag = hasMatchingTagInSpecs(specs, tag);
+        if (hasTag) {
+          foundTags.add(tag);
+        }
+      }
+
+      // Map found tags to their corresponding runMatrixEntries and pwGrepTags
+      for (const tag of foundTags) {
+        const runMatrixEntries = tagToRunMatrixMap[tag] || [];
+        // Only include entries that are in the module's run matrix
+        const relevantEntries = runMatrixEntries.filter((entry) => moduleRunMatrix.includes(entry));
+
+        for (const runMatrixEntry of relevantEntries) {
+          const pwGrepTag = getPlaywrightGrepTag(runMatrixEntry);
+          const serverMode = normalizeServerMode(runMatrixEntry);
+          const group = data.group as string;
+
+          // Skip platform group for serverless mode
+          // Serverless should only have solutions: search, security, observability, etc.
+          if (serverMode === 'serverless' && group === 'platform') {
+            continue;
+          }
+
+          // Determine the Playwright project name based on server mode
+          // For cloud runs: stateful -> 'ech', serverless -> 'mki'
+          const project = serverMode === 'stateful' ? 'ech' : 'mki';
+
+          // Initialize structure if needed
+          if (!result[serverMode]) {
+            result[serverMode] = {};
+          }
+          if (!result[serverMode][group]) {
+            result[serverMode][group] = {
+              project,
+              tag: pwGrepTag,
+              configs: [],
+            };
+          }
+
+          // Add config if not already present
+          // Note: project and tag are set on first initialization and not overwritten
+          // This ensures consistent project and tag per group+serverMode combination
+          if (!result[serverMode][group].configs.includes(pwConfigPath)) {
+            result[serverMode][group].configs.push(pwConfigPath);
+          }
+        }
+      }
+    }
+  }
+
+  const endTime = Date.now();
+  const durationSecs = ((endTime - startTime) / 1000).toFixed(2);
+  log.info(
+    `Filtering completed in ${durationSecs} seconds. Playwright configs with tests filtered.`
```

Comment:
The command is taking around 40 seconds on my machine. Would it be more efficient to take a look at the list of enabled plugins/packages in `scout_ci_config.yml`, and search the configs from within those test file paths only?